app/models/cookbook.rb

```diff
@@ -407,4 +401,23 @@ def deprecate_search(query)
   def copy_name_to_lowercase_name
     self.lowercase_name = name.to_s.downcase
   end
+
+  def create_new_collaborator(initiator)
+    unless collaborators.where(user_id: initiator.id).present?
+      collaborator = Collaborator.new(user_id: initiator.id, resourceable: self)
+      collaborator.save!
+    end
+  end
+
+  def delete_old_collaborator(recipient)
+    if collaborator_users.include?(recipient)
+      collaborator = collaborators.where(
+        user_id: recipient.id,
+        resourceable: self
+      ).first
+      # Do not destroy collaborator is collaborator does not exist
+      # OR if the collaborator is associated with a group
+      collaborator.destroy unless collaborator.nil? || collaborator.group_id.present?
+    end
```

Comment:
Based on the `unless`, it looks like we want to find a non-group-affiliated `Collaborator` for the `recipient` user and delete that one. More tinkering:

```
collaborator = collaborators.where(
  user_id: recipient.id,
  group_id: nil
).first
collaborator.destroy if collaborator
```

Also, in staring at these methods, I note that from the private methods' perspectives, they do not care that the parameter happens to be an `initiator` or a `recipient`. The param is just a `user`. The context of what kind of user only matters up in the body of the `transfer_ownership` method.